evm/src/main/java/org/hyperledger/besu/evm/UInt256.java

```diff
@@ -330,21 +384,15 @@ private static int nSetLimbs(final int[] x) {
   }
 
   private static int compareLimbs(final int[] a, final int aLen, final int[] b, final int bLen) {
-    int cmp;
-    if (aLen > bLen) {
-      for (int i = aLen - 1; i >= bLen; i--) {
-        cmp = Integer.compareUnsigned(a[i], 0);
-        if (cmp != 0) return cmp;
-      }
-    } else if (aLen < bLen) {
-      for (int i = bLen - 1; i >= aLen; i--) {
-        cmp = Integer.compareUnsigned(0, b[i]);
-        if (cmp != 0) return cmp;
-      }
+    if (aLen != bLen) {
```

Comment:
This is incorrect I think: a = <1, 0, 0, 0> with length 4 is smaller than b = <1, 1, 0, 0> with length 2 (4 limbs but extrapolate to 8 limbs if you want). Yet both are valid internal representations of UInt256.
We have to revert back this change.